services/apps/src/apps/service/apps/jobs/sharing.clj

```diff
@@ -173,16 +173,28 @@
    (catch ce/clj-http-error? {:keys [body]}
      (str "unable to unshare result folder: " (:error_code (service/parse-json body))))))
 
-;; The apps client isn't used at this time, but it will be once we extend analysis sharing
-;; to HPC apps.
+(defn- unshare-input-file
+  [sharer sharee path]
+  (try+
+   (data-info/unshare-path sharer path sharee)
```

Comment:
Sounds good. :shipit:

As a sort of aside, I don't know if it would make sense to consider the permission the file currently has as a marker of whether unsharing should happen -- for example, if the sharee has write permission but analysis sharing would only have given them read, it could be argued they shouldn't have their permissions removed. But this could get confusing for other reasons and I'm not sure what variant is more confusing.

We could also do something like have analysis sharing add an administrative AVU when the user already had permissions before the sharing, so that the unsharing endpoint can restore those permissions exactly. But that's definitely a task for later if we want to do it!